### PR TITLE
dev/core#2486 Add survey v4 api

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#2486 Add survey v4 api

Before
----------------------------------------
Survey api missing for v4

After
----------------------------------------
Tada

Technical Details
----------------------------------------
@colemanw I created this to help with testing out your patch. I defined the option group pseudoconstant but it is pretty fugly & I don't know it helps much since with it configured I get this ( I think that is without prefetch set to false).

I also notice there is one html field that is rendered as html - ideally we would detect that & not show it in all it's taggy glory.

image

Given this is a barely used api I don't want to put too much into it - it's just a question of whether I do or don't add a not-that-great pseudoconstant on the result fields at this point

I guess tests will fail

Comments
----------------------------------------

### CODE REVIEW

File: xml/schema/Campaign/Survey.xml [hunk 1]
```diff
@@ -214,6 +214,13 @@
     <default>NULL</default>
     <comment>Used to store option group id.</comment>
```
Comment (colemanw) [inline xml/schema/Campaign/Survey.xml:215]: @eileenmcnaughton I'm pretty sure the comment here is incorrect. This field appears to be used to store the option **value** of the option group configured for this survey.
Screenshot from 2021-09-02 21-21-58

Comment (eileenmcnaughton) [reply xml/schema/Campaign/Survey.xml:215]: @colemanw I created a survey & got a value of 106 in that field - it correlated to the option group id - then when you phone people up you create activities & they have a field (result?) that maps to the option value

Comment (colemanw) [reply xml/schema/Campaign/Survey.xml:215]: Oh, no, wait, I get it. This stores the option_group_id of the option group being displayed on this form.

File: xml/schema/Campaign/Survey.xml [hunk 2]
```diff
@@ -214,6 +214,13 @@
...
     <add>3.3</add>
+    <pseudoconstant>
+      <table>civicrm_option_group</table>
+      <keyColumn>id</keyColumn>
+      <labelColumn>title</labelColumn>
+      <nameColumn>name</nameColumn>
+      <prefetch>FALSE</prefetch>
```
Comment (colemanw) [inline xml/schema/Campaign/Survey.xml:222]: I checked the core code and it uses a pretty dumb hack which we might as well do here too (and then no need for disabling prefetch):

```suggestion
      <condition>name LIKE "civicrm_survey_%"</condition>
```

Comment (colemanw) [reply xml/schema/Campaign/Survey.xml:222]: This is the dumb hack I'm referring to: https://github.com/civicrm/civicrm-core/blob/0419bf7b41a6c5b5ac22a31bf0b0589dc649e1b3/CRM/Campaign/BAO/Survey.php#L694

Comment (eileenmcnaughton) [reply xml/schema/Campaign/Survey.xml]: @colemanw ok - I've made that change

**Comment by eileenmcnaughton** [comment]
This caused singleValueAlter test to start failing - I just added Survey to the skip line. That test has gotten really slow & doesn't add the value it did when our code was more chaotic

**Pull request merged**